ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java

```diff
@@ -105,8 +111,10 @@ public List<TaskDescriptor> planTasks() {
     if (this.tasks != null) {
       return this.tasks;
     }
-    long startTime = System.nanoTime();
-
+    SelfOptimizingPlanDurationReport selfOptimizingPlanDurationReport =
+        new SelfOptimizingPlanDurationReport(tableRuntime.getTableIdentifier().toString());
+    Timer timer = selfOptimizingPlanDurationReport.tableOptimizingPlanDuration();
+    Timer.Context context = timer.time();
```

Comment:
I think we can use selfOptimizingPlanDurationReport.tableOptimizingPlanDuration().time(planTasks()) to collect statistics. This way, we don't have to invade the internal logic of the planTasks() method. What do you think?

Reply:
> I think we can use selfOptimizingPlanDurationReport.tableOptimizingPlanDuration().time(planTasks()) to collect statistics. This way, we don't have to invade the internal logic of the planTasks() method. What do you think?

How to understand this logic, is it also counted in the plan cost?

```
    if (this.tasks != null) {
      return this.tasks;
    }
```

Reply:
Perhaps we should separate the logic of `return this.tasks;` and actually planting the task. For example:
`public List<TaskDescriptor> findTask() {
    if (this.tasks != null) {
      return this.tasks;
    } else {
      Timer timer = selfOptimizingPlanDurationReport.tableOptimizingPlanDuration();
      timer.time(planTasks());
      metricsReporters.report(selfOptimizingPlanDurationReport);
      return this.tasks;
   }
}`

Reply:
> Perhaps we should separate the logic of `return this.tasks;` and actually planting the task. For example:
> 
> ```
> public List<TaskDescriptor> findTask() {
>     if (this.tasks != null) {
>       return this.tasks;
>     } else {
>       Timer timer = selfOptimizingPlanDurationReport.tableOptimizingPlanDuration();
>       timer.time(planTasks());
>       metricsReporters.report(selfOptimizingPlanDurationReport);
>       return this.tasks;
>    }
> }
> ```
 +1